src/gui/title_bar.cpp

```diff
@@ -0,0 +1,600 @@
+		if (iequals(setting, "cycles")) {
+			check_not_set_cycles();
+			config.show_cycles = true;
+			continue;
+		}
+
+		if (iequals(setting, "dosbox")) {
+			check_not_set_dosbox();
+			config.show_dosbox = true;
+			continue;
+		}
+
+		if (iequals(setting, "program") ||
+		    iequals(setting, "program=name")) {
+			check_not_set_program();
+			config.program = ProgramDisplay::Name;
+			continue;
+		}
+
+		if (iequals(setting, "program=path")) {
+			check_not_set_program();
+			config.program = ProgramDisplay::Path;
+			continue;
+		}
+
+		if (iequals(setting, "program=segment")) {
+			check_not_set_program();
+			config.program = ProgramDisplay::Segment;
+			continue;
+		}
+
+		if (iequals(setting, "version") ||
+		    iequals(setting, "version=simple")) {
+			check_not_set_version();
+			config.version = VersionDisplay::Simple;
+			continue;
+		}
+
+		if (iequals(setting, "version=detailed")) {
+			check_not_set_version();
+			config.version = VersionDisplay::Detailed;
+			continue;
+		}
+
+		LOG_WARNING("SDL: Unknown setting '%s' in 'titlebar_content'",
+		            setting.c_str());
+	}
+
+	config.titlebar_text = conf->Get_string("titlebar_text");
+	trim(config.titlebar_text);
+}
+
+void TITLEBAR_AddConfig(Section_prop &secprop)
+{
+	constexpr auto always = Property::Changeable::Always;
+
+	Prop_string* prop_str = nullptr;
+
+	prop_str = secprop.Add_string("titlebar_content", always,
+	                              "animation cycles dosbox program=name");
+	prop_str->Set_help(
+	        "Space separated list of information to be displayed on the window title bar\n"
+	        "('animation cycles dosbox program=name' by default). Possible information are:\n"
+	        "  animation:          Animates title bar audio/video recording mark. Results\n"
+	        "                      might depend on your screen font.\n"
+	        "  cycles:             Show CPU cycles setting.\n"
+	        "  dosbox:             DOSBox emulator name.\n"
+	        "  program[=<value>]:  Display the name of a running program instead of\n"
+	        "                      'titlebarbar_text'. The <value> can be one of:\n"
+	        "                        name:     Program name, with file extension (default).\n"
+	        "                        path:     Name, extension, and full absolute path.\n"
+	        "                        segment:  Displays program memory segment name.\n"
+	        " 	                  Note: With some software (like Windows 3.1x in enhanced\n"
+	        "                      mode) it is impossible to recognize the full program name\n"
+	        "                      or path; in such case 'segment' value is used instead.\n"
+	        "  version[=<value>]:  Display DOSBox version information after emulator name.\n"
+	        "                      The <value> can be one of:\n"
+	        "                        simple:    Simple version format (default).\n"
+	        "                        detailed:  Include GIT hash, if available.");
```

Comment:
These params should be listed in the order they appear in in the titlebar.

Reply:
Changed.